src/metabase/sync/analyze/fingerprint/sample.clj

```diff
@@ -0,0 +1,20 @@
+(ns metabase.sync.analyze.fingerprint.sample
+  "Analysis sub-step that fetches a sample of values for a given Field, which is used to generate a fingerprint for it.
+   Currently this is dumb and just fetches a contiguous sequence of values, but in the future we plan to make this
+   more sophisticated and have different types of samples for different Fields."
+  (:require [metabase.driver :as driver]
+            [metabase.models.table :refer [Table]]
+            [metabase.sync.interface :as i]
+            [schema.core :as s]
+            [toucan.db :as db]))
+
+(s/defn ^:always-validate basic-sample :- (s/maybe i/ValuesSample)
```

Comment:
we'll probably eventually want a flag in this function that turns it from a sample to a "full sample". This mucks with the naming of this space, and isn't vital right this second, but would probably reduce future work.